File: app/src/main/java/com/garymcgowan/moviepedia/model/MovieRepository.java
package com.garymcgowan.moviepedia.model;

import java.util.List;

import io.reactivex.Flowable;
import io.reactivex.Single;

/**
 * Created by garymcgowan on 08/05/2017.
 */

public interface MovieRepository {

    Flowable<List<Movie>> getMovieSearch(String search);

    Single<Movie> getMovieDetails(String movieId);
}

File: app/src/main/java/com/garymcgowan/moviepedia/model/MovieRepository.kt
package com.garymcgowan.moviepedia.model

import io.reactivex.Flowable
import io.reactivex.Single

interface MovieRepository {

    fun getMovieSearch(search: String): Flowable<List<Movie>>

    fun getMovieDetails(movieId: String): Single<Movie>
}
